haskell/test/assets/storage_with_metadata.tz

```diff
@@ -0,0 +1,62 @@
+(Pair (Pair (Pair (Pair (Pair (Pair (Pair 5 10) 1 "KT1PWx2mnDueood7fEmfbBDKx1D9BAnnXitn")
```

Comment:
Shouldn't this file be generated on the fly (like the other .tz files), instead of being stored in the repo?

Reply:
The problem with that is we have to build and run haskell executable to be able to generate this file. So it is quite tricky to generate on the fly, I'm really not sure how to do it so I put a constant file for now.

Reply:
Ah, I see the problem.

Well, we could run the haskell executable via nix (like we do [in the pipeline to generate the typescript bindings](https://github.com/serokell/segmented-cfmm/blob/d1a0e588b56916320487200e58327c9bd2c79d6d/.buildkite/pipeline.yml#L50-L55)), but the problem here is that the haskell executable is being invoked _from the makefile_.

I guess we'd need to modify the [`build-ligo` nix component](https://github.com/serokell/segmented-cfmm/blob/d1a0e588b56916320487200e58327c9bd2c79d6d/ci.nix#L92-L98) to
1. invoke `packages.segmented-cfmm.exes.segmented-cfmm` with `--print-metadata-byte`
2. invoke `packages.segmented-cfmm.exes.segmented-cfmm` with `--generate-metadata-uri`
3. `make $(OUT)/metadata_map`

Not sure how to make a derivation call a different nix target, or if this is worth doing tho... @pasqu4le, what do you think?

Reply:
The problem with that is that `packages.segmented-cfmm` depends on `build-ligo` (see `preBuild`) which would make this circular...

I think also that we would have the same problem running this locally BTW and a solution would be needed for running this with `stack` and/or `make`.

I believe that the only proper solution would be to move the metadata generation to a different package, but I think that we should not do this here and accept this workaround for the time being.

Reply:
Cool, I'm approving this then!